### ISSUE TITLE
YouTube video controls very slow when extension is enabled

### ISSUE THREAD

**Issue by bertvandepoel** [mooltipass/extension#111]
I've noticed for several years now that the YouTube video controls have been very slow. I always presumed it was either because of a lack of Firefox GPU acceleration on Linux or some other Linux related issue. However, the actual culprit seems to be the Mooltipass extension.

When the Mooltipass extension is disabled, i notice a distinct delay of up to 5 seconds when I click in the middle of a video to pause it. There's also a delay when interaction with the cog to for example change the quality, up to the point where it can take me half a minute or more to force the quality onto 720p because each step takes a while to load and then another while to become interactive (if I click before my pointer changes, I have to start over).

I'm guessing this is caused by some weird interaction between the addon and the control elements, but it's quite frustrating if I'm honest. I would suggest to help debug and fix this, but I have no clue at all would could even cause this within the Mooltipass extension.

**Comment by limpkin** [comment]
good news is that we're about to release that fixes this very issue :)
hopefully next week :)

**Comment by limpkin** [comment]
@bertvandepoel the extension is still under review on mozilla's side...

**Comment by bertvandepoel** [comment]
I had been wondering about that :P Surprised they're taking a long time. I was under the impression they mostly checked initial releases and large rewrites, or is this the latter?

**Comment by chasgames** [comment]
I have the same, but for Google Chrome, random video buffering and unresponsive video controls. Once the extension is disabled it's responsive again.

**Comment by limpkin** [comment]
@bertvandepoel oh no....  :/. Still being reviewed :/

**Comment by principis** [comment]
it's here :tada:

**Comment by bertvandepoel** [comment]
Got the update a few hours ago and the difference is night and day <3
Will test it for a few more days and close the issue afterwards ;)

**Comment by kbembedded** [comment]
I've seen massive performance improvements in FireFox too with the update. 👍

**Comment by limpkin** [comment]
awesome!!!

**Comment by chasgames** [comment]
@limpkin Did the update also get pushed to Google Chrome users? I tested today but still notice performance increase when disabling the extension on youtube

**Comment by limpkin** [comment]
that's odd... we never got any complaint since we pushed our latest version :/.
@chasgames could you give it another go?

**Comment by bertvandepoel** [comment]
I've been wanting to get back to this issue. So sometimes I do still notice the slowdowns. But it's weird, it's a specific video, when I open a new tab with another video, it's often gone. I wanted to try and find the reason behind this problem before reporting, since I'm not even completely sure it's caused by the Mooltipass extension. But yeah, I do sometimes see random slowdowns when interacting with Youtube video elements (but not the page, just like changing the video quality and stuff like that).

**Comment by chasgames** [comment]
Yep @limpkin I can still replicate it, still noticeable on chromium and the blacklist workaround works every time for me.

As a base line I switched from google chrome to chromium, clean install with only the mooltipass extension, disabled cache in developer tools, and it still does it. Other than the random slow downs, I can also notice immediately:

- Spam right click on any video, the context menu for the youtube video player will show up around 8 - 14 right clicks, browser context menu mostly always shows but sometimes nothing happens.
- Spam left click on the settings cog wheel on any video,  it seems to be delayed, no animation and then can go unresponsive.
- Theater mode and full screen mode lag into their mode.
- Sluggish loading of new thumbnails when scrolling down the page


Mootipass with "youtube.com" added to the blacklist: 
+Every right click it alternates between the youtube control context menu and the browser context menu, very responsive.
+Every left click on the settings cog loads the animation and shows the menu, very responsive.
+Theater mode and full screen instantly go straight into their mode.
+Thumbnails load nearly instantly
+No rare slow down / hangs

Unplugging the Mooltipass BLE didn't make any difference, Firefox seems to handle it a lot better can barely detect much difference.

**Comment by limpkin** [comment]
@chasgames beta version 1.9.10 with a possible fix is about to be released for chrome at https://chrome.google.com/webstore/detail/mooltipass-extension/mkjlelalgdinanmcljpgkojjolkdcebh?hl=en&authuser=0 , could you give it a go?

**Comment by chasgames** [comment]
I tried it but I'm afraid no change, also tried on linux machine with chrome but the issue can be replicated on 1.10.0 immediately

**Comment by bertvandepoel** [comment]
I've been running 1.10.0 on Firefox for a few days now. While it initially seemed fine, I just had it happen again. Though after leaving that video paused for a bit, the problem disappeared. Still quite confused why this is such a persistent issue.

**Comment by kbembedded** [comment]
Adding another potential detail to this: I don't see it when just browsing random videos; however, I do see the slowdown start to occur when watching a playlist over a handful of videos. I've not done comparison testing with and without the extension running, but that might be something to consider doing if others are mostly watching playlists and seeing it too?

Windows 10, FF 94.0.1, extension 1.10.1

**Comment by bertvandepoel** [comment]
Any news about this issue? It really remains an extremely frustrating bug.

**Comment by bertvandepoel** [comment]
I noticed a very similar extreme slowdown when trying to create an event on linkedin (something I almost never do). I had to disable the extension for it to work. Now, I also run JShelter, and perhaps that addon was just acting up, but it seemed potentially relevant.

**Comment by drwonky** [comment]
Ahh, I hadn't associated the play/pause slowdown with Mooltipass Firefox extension, however there is another insidious problem that's probably related:  When reloading my subscription feed I get high CPU usage on a single thread and the video preview images take 20s to load while it's gobbling CPU.  I discovered this was caused by the Mooltipass extension by using the performance tab in developer tools.  If this is not associated with this specific issue, I'd be happy to open another issue.  However, if this is caused by the same underlying issue in the extension, I hope this information helps other people find a cause too.

**Comment by bertvandepoel** [comment]
I fear that the problems that arise from the extension on YouTube are underreported because everyone I've talked to has just disabled the plugin for youtube.com using the blacklist functionality. I wouldn't be surprised if both issues were related.
